server/monitor/src/main/java/org/apache/accumulo/monitor/next/InformationFetcher.java

```diff
@@ -0,0 +1,355 @@
+import org.apache.accumulo.core.rpc.ThriftUtil;
+import org.apache.accumulo.core.rpc.clients.ThriftClientTypes;
+import org.apache.accumulo.core.trace.TraceUtil;
+import org.apache.accumulo.core.util.UtilWaitThread;
+import org.apache.accumulo.core.util.threads.ThreadPools;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.thrift.transport.TTransportException;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.eclipse.jetty.util.NanoTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import com.github.benmanes.caffeine.cache.Scheduler;
+import com.google.common.net.HostAndPort;
+
+public class InformationFetcher implements RemovalListener<ServerId,MetricResponse>, Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(InformationFetcher.class);
+
+  public static class InstanceSummary {
+    private final String instanceName;
+    private final String instanceUUID;
+    private final Set<String> zooKeepers;
+    private final Set<String> volumes;
+    private final String version;
+
+    public InstanceSummary(String instanceName, String instanceUUID, Set<String> zooKeepers,
+        Set<String> volumes, String version) {
+      super();
+      this.instanceName = instanceName;
+      this.instanceUUID = instanceUUID;
+      this.zooKeepers = zooKeepers;
+      this.volumes = volumes;
+      this.version = version;
+    }
+
+    public String getInstanceName() {
+      return instanceName;
+    }
+
+    public String getInstanceUUID() {
+      return instanceUUID;
+    }
+
+    public Set<String> getZooKeepers() {
+      return zooKeepers;
+    }
+
+    public Set<String> getVolumes() {
+      return volumes;
+    }
+
+    public String getVersion() {
+      return version;
+    }
+  }
+
+  private class MetricFetcher implements Runnable {
+
+    private final ServerContext ctx;
+    private final ServerId server;
+    private final SystemInformation summary;
+
+    private MetricFetcher(ServerContext ctx, ServerId server, SystemInformation summary) {
+      this.ctx = ctx;
+      this.server = server;
+      this.summary = summary;
+    }
+
+    @Override
+    public void run() {
+      try {
+        Client metricsClient = ThriftUtil.getClient(ThriftClientTypes.METRICS,
+            HostAndPort.fromParts(server.getHost(), server.getPort()), ctx);
+        try {
+          MetricResponse response = metricsClient.getMetrics(TraceUtil.traceInfo(), ctx.rpcCreds());
+          summary.processResponse(server, response);
+        } finally {
+          ThriftUtil.returnClient(metricsClient, ctx);
+        }
+      } catch (Exception e) {
+        LOG.warn("Error trying to get metrics from server: {}. Error message: {}", server,
+            e.getMessage());
```

Comment:
Might be useful to see the actual stack trace if there's a warning/error here:

```suggestion
        LOG.warn("Error trying to get metrics from server: {}", server, e);
```